dotCMS/src/main/java/com/dotcms/business/SystemTableFactoryImpl.java

```diff
@@ -34,23 +35,16 @@ protected Optional<String> find(final String key) throws DotDataException {
         Object value = null;
         if(UtilMethods.isSet(key)) {
 
-            value = this.systemCache.get(key);
-            if(Objects.isNull(value)) {
+            value = this.systemCache.getOrUpdate(key, ()-> {
 
-                final List<Map<String, Object>> result = new DotConnect()
+                final List<Map<String, Object>> result = Try.of(()->new DotConnect()
                         .setSQL(" SELECT * FROM system_table WHERE key = ? ")
                         .addParam(key)
-                        .loadObjectResults();
+                        .loadObjectResults()).getOrElse(Collections.emptyList());
```

Comment:
I think we should throw here if there is a DB connection.  If there is an issue with the query and this is in a transaction, it will break the transaction.

Reply:
Done thanks